validations/src/validations.rs

```diff
@@ -1514,20 +1552,20 @@ mod tests {
         let mut counter = Counter::new(7);
         counter.increment(6);
         assert_eq!(counter.max_val, 1);
-        assert_eq!(counter.max_pos, 6);
+        assert_eq!(counter.max_pos, Some(6));
 
         counter.increment(6);
         assert_eq!(counter.max_val, 2);
-        assert_eq!(counter.max_pos, 6);
+        assert_eq!(counter.max_pos, Some(6));
 
         counter.increment(0);
         assert_eq!(counter.max_val, 2);
-        assert_eq!(counter.max_pos, 6);
+        assert_eq!(counter.max_pos, Some(6));
 
         counter.increment(0);
```

Comment:
You can add a test for ties after this line:

    assert_eq!(counter.max_pos, None);